### ISSUE TITLE
Workspace list shows invisible workspaces when next visible is created

### ISSUE THREAD

**Issue by martyngigg** [mantidproject/mantid#6798]
This issue was originally [TRAC 5952](http://trac.mantidproject.org/mantid/ticket/5952)

To reproduce:
* Go to View->Preferences->Mantid->Options and uncheck "show invisible workspaces"
* Click Load;
* Pick a file and prefix the suggested workspace name with ```__```
* The algorithm will run but there should be nothing in the list
* Click load again and run the same thing but without the prefix. The invisible workspace will appear alongside the visible after the algorithm completes

**Comment by FedeMPouzols** [comment]
This is still an issue. If you go to Edit->Preferences->Mantid->Options and set "Show invisible workspaces", then "apply", you won't see them. If for example you reload the invisible workspace as invisible ("_") then you will see it.

**Comment by FedeMPouzols** [comment]
@NickDraper: this could be another good "anyone" ticket for new starters.

**Comment by PranavBahuguna** [comment]
Could you briefly explain how in the code, checking/unchecking the "Show invisible workspaces" option causes the workspaces dock (MantidDock.cpp) to hide or unhide workspaces in the first place?

**Comment by PranavBahuguna** [comment]
On the previous commit, I have changed `ConfigDialog::updateMantidOptionsTab` to have `MantidDock::updateTree` called - the method responsible for hiding or unhiding workspaces. I have noticed however that `updateTree()` is intended to only be called when the GUI thread is free, and I assume this means that no other windows like the preferences window could be open when `updateTree()` is called. Is there any good reason why this is so? Without this restriction, I could hide/unhide workspaces on clicking the apply button in a similar manner as to algorithm categories in the dock can be made visible or not by toggling their checkboxes in the Preferences window,
